### PR TITLE
Convert script/configure-web-proxy from POSIX sh to bash

### PR DESCRIPTION
**Pull request by asdil12**

### CODE REVIEW

File: script/configure-web-proxy [hunk 1]
```diff
@@ -27,11 +27,11 @@ done
 
 sed -i -e 's/^.*httpsonly.*$/httpsonly = 0/g' /etc/openqa/openqa.ini
 
-if [ "$web_proxy" = "nginx" ] ; then
+if [[ $web_proxy == "nginx" ]] ; then
 	echo "Setting up nginx"
 	sed "s/openqa.example.com/$(hostname)/" /etc/nginx/vhosts.d/openqa.conf.template > /etc/nginx/vhosts.d/openqa.conf
 	sed -i -e "s/\(^[^#]*server_name  localhost;\)/#\1/" /etc/nginx/nginx.conf
-elif [ "$web_proxy" = "apache" ] || [ "$web_proxy" = "apache2" ] ; then
+elif [[ $web_proxy == "apache" ]] || [[ $web_proxy == "apache2" ]] ; then
```
Comment (perlpunk) [inline script/configure-web-proxy:34]:
```suggestion
elif [[ $web_proxy == "apache" || $web_proxy == "apache2" ]] ; then
```

**Review by okurz — COMMENTED** [review]
Why not keep it POSIX sh? Or was it inconsistent in before?

**Comment by Martchus** [comment]
Because almost all systems have bash installed (even if it is not the shell used by the user) so it makes no sense to limit ourselves to inferior shell syntax.

**Comment by okurz** [comment]
> Because almost all systems have bash installed (even if it is not the shell used by the user) so it makes no sense to limit ourselves to inferior shell syntax.

Ok, that's a good reason why the change can be done without significant impact but is not a reason why the change would provide a benefit

**Comment by asdil12** [comment]
> > Because almost all systems have bash installed (even if it is not the shell used by the user) so it makes no sense to limit ourselves to inferior shell syntax.
> 
> Ok, that's a good reason why the change can be done without significant impact but is not a reason why the change would provide a benefit

I guess we should decide: Make all scripts rely on bash features like `==` or `[[` or be posix sh compliant.
I see no value in the current chaotic situation where we try to enforce `[[` in one file but hesitate to do the same in other files.

**Review by Martchus — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Comment by okurz** [comment]
Missing codecov feedback

@Mergifyio rebase

**Comment by okurz** [comment]
I think mergify never rebased but it seems the codecov report nevertheless eventually arrived :D

**Pull request merged**